robot/src/robot.py

```diff
@@ -18,9 +19,48 @@ def __init__(self):
 
         self.ultrasonic_sensor = UltrasonicSensor(Port.S4)
 
+        self.routes = {1: [("forward", 300), ("rotate", 90), ("forward", 300), ("rotate", 90),
+                      ("forward", 300), ("rotate", 90), ("forward", 300), ("rotate", 90)]}
+
+
+    def set_movement_speed(self, speed):
+        self.drive_base.settings(speed=speed)
+
+    # Get the distance traveled by the robot in centimeters
+    def distance_travelled(self):
+        return self.drive_base.distance() / 10
+
     def obstacle_in_front(self):
         return self.ultrasonic_sensor.distance() > 300
 
+    def stop_moving(self):
+        self.drive_base.stop()
+
+    # Direction == 1 is forward, direction == -1 is backward
+    def move(self, direction):
+        if direction == 1:
+            self.drive_base.drive(200, 0)
+        elif direction == -1:
+            self.drive_base.drive(-200, 0)
+        else:
+            self.stop_moving()
+
+    # Positive distance is forward, negative distance is backward
+    def move_distance(self, distance):
+        self.drive_base.straight(distance)
+
+    # Positive angle is clockwise, negative angle is counterclockwise
+    def rotate(self, angle):
+        self.drive_base.turn(angle)
+
+    async def follow_route(self, route_id):
+        route = self.routes.get(route_id)
+        for command in route:
+            if command[0] == "forward":
+                self.drive_base.straight(command[1])
```

Comment:
Use move_distance here
```suggestion
                self.move_distance(command[1])
```